plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/UnsafeIntrinsics.java

```diff
@@ -264,15 +266,15 @@ private static void registerGetAndModIntrinsics(final CompilationContext ctxt) {
                 "BitwiseXor", BasicBlockBuilder::getAndBitwiseXor,
                 "Set", (BuilderGetAndModOp) BasicBlockBuilder::getAndSet
             ).entrySet()) {
-                for (Map.Entry<String, MemoryAtomicityMode> suffixAndMode : Map.of(
-                    "", VOLATILE,
-                    "Acquire", ACQUIRE,
-                    "Release", RELEASE
+                for (Map.Entry<String, AccessMode[]> suffixAndMode : Map.of(
+                    "", new AccessMode[] { GlobalAcquire, GlobalRelease },
+                    "Acquire", new AccessMode[] { SingleAcquire, SingleOpaque },
+                    "Release", new AccessMode[] { SinglePlain, SingleRelease }
```

Comment:
Going off of `VarHandle` again, I think these would need to be:
* `GlobalSeqCst`, `GlobalSeqCst`
* `GlobalAcquire`, `*Plain`
* `*Plain`, `GlobalRelease`